src/main/java/duke/services/Parser.java

```diff
@@ -0,0 +1,155 @@
+                descBuilder.append(" ");
+            } else {
+                descBuilder.append(words[currWordIndex]).append(" ");
+                emptyDesc = false;
+            }
+            ++currWordIndex;
+        }
+
+        if (emptyDesc) {
+            throw new IllegalArgumentException("OOPS!!! Description can't be empty");
+        }
+
+        return descBuilder.deleteCharAt(descBuilder.length()-1).toString();
+    }
+
+    /**
+     * Retrieves the timing argument in the command, which must be of the form d/M/yyyy followed by an optional (h:mm)am/pm
+     * @param words The words of the command entered, first is some valid command name
+     * @param flag The flag that the timing belongs to
+     * @return The timing specified in words
+     * @throws IllegalArgumentException If words is missing flag or the timing is empty/incorrect format
+     */
+    public static String getTiming(String[] words, String flag) {
+        if (currWordIndex >= words.length) {
+            throw new IllegalArgumentException("OOPS!!! " + flag + " not found");
+        } else if (currWordIndex == words.length - 1) {
+            throw new IllegalArgumentException("OOPS!!! Timing for " + flag + " can't be empty");
+        }
+
+        ++currWordIndex;
+        try {
+            return (currWordIndex == words.length - 1)
+                    ? reformatDate(words[currWordIndex], "d/M/yyyy", "d MMM yyyy")
+                    : reformatDateTime(words[currWordIndex] + " " + words[++currWordIndex],
+                    "d/M/yyyy h:mma", "d MMM yyyy, h:mma");
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("OOPS!!! I don't understand that date or time");
+        }
+    }
+
+    public static String reformatDate(String date, String inFormat, String outFormat) {
+        DateTimeFormatter formatter = new DateTimeFormatterBuilder()
+                .appendPattern(inFormat)
+                .toFormatter(Locale.getDefault());
+        LocalDate ld = LocalDate.parse(date, formatter);
+        return ld.format(DateTimeFormatter.ofPattern(outFormat, Locale.getDefault()));
+    }
+
+    public static String reformatDateTime(String dateTime, String inFormat, String outFormat) {
+        DateTimeFormatter formatter = new DateTimeFormatterBuilder()
+                .parseCaseInsensitive()
+                .appendPattern(inFormat)
+                .toFormatter(Locale.getDefault());
+        LocalDateTime ldt = LocalDateTime.parse(dateTime, formatter);
+        return ldt.format(DateTimeFormatter.ofPattern(outFormat, Locale.getDefault()));
+    }
+
+    /**
+     * Gets the task number (integer pointing to a task) specified in the command
+     *
+     * @param words The words of the command entered, first is always some valid command name
+     * @return The task number specified
+     * @throws IllegalArgumentException If the argument(s) supplied in words isn't an integer from 1 to the number of stored tasks
```

Comment:
It might help to shorten line 135 by making the description include the information instead.
Loving the helpful javadoc though! 
```suggestion
     * @throws IllegalArgumentException If size of words[] is invalid.
     ```